### PR TITLE
[BB-960] Don't fail when Github user does not exist

### PR DESCRIPTION
**Pull request by viadanna**
Description
=========

This PR implements a check for the Github users before provisioning the servers to prevent failures when a user isn't found.

Testing
----------

1. Check out this branch in Stage and restart the server.
2. Replace a user's Github handle with one that does not exist.
3. Start provisioning a new App Server.
4. Make sure it does not fail when adding users, as the user that does not map to a Github user does not show the `github` key.

```
2019-06-25 10:29:07-0300 INFO
changed: [54.38.70.171] => (item={u'type': u'admin', u'name': u'viadannap'})
2019-06-25 10:29:07-0300 INFO
changed: [54.38.70.171] => (item={u'github': True, u'type': u'admin', u'name': u'viadanna'})
```

### CODE REVIEW

File: instance/models/mixins/openedx_config.py [hunk 1]
```diff
@@ -383,12 +384,20 @@ def _get_configuration_variables(self):
...
+                username for username in self.admin_users
+                if session.get('https://github.com/{}.keys'.format(username)).status_code == 200
+            ]
             template.update({
                 "COMMON_USER_INFO": [
                     {
                         "name": github_username,
                         "github": True,
                         "type": "admin"
+                    } if github_username in users else {
```
Comment (lgp171188) [inline instance/models/mixins/openedx_config.py]: @viadanna, do we have to include the users with missing keys as they will not have a way to login?

File: instance/models/utils.py [hunk 2]
```diff
@@ -651,3 +653,18 @@ def _is_json_serializable(obj):
...
+    session = session or requests.Session()
+    retry = Retry(
+        total=retries,
+        read=retries,
+        connect=retries,
+        backoff_factor=backoff_factor,
+        status_forcelist=status_forcelist,
+    )
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('http://', adapter)
```
Comment (lgp171188) [inline instance/models/utils.py]: @viadanna, is there a reason we need HTTP?

File: instance/models/mixins/openedx_config.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 from django.conf import settings
```
Comment (lgp171188) [inline instance/models/mixins/openedx_config.py]: @viadanna, can you add unit tests for the changes made in this PR?

Comment (viadanna) [reply instance/models/mixins/openedx_config.py]: Sure :+1:

File: instance/models/mixins/openedx_config.py (hunk 1 above)
Comment (viadanna) [reply instance/models/mixins/openedx_config.py]: I thought it was better than just ignoring the user. I can remove this instead.

File: instance/models/utils.py (hunk 2 above)
Comment (viadanna) [reply instance/models/utils.py]: Not for Github, I can remove it but it'll make the feature less generic.

**Pull request merged**